docker/Dockerfile

```diff
@@ -12,43 +12,267 @@
+RUN PL_BACKEND=lightning_kokkos python setup.py bdist_wheel
+
+# Install python3 and setup runtime virtual env in CUDA-12-runtime image (includes CUDA runtime and math libraries)
+# Install lightning-kokkos CUDA backend
+FROM nvidia/cuda:12.2.0-base-ubuntu22.04 AS wheel-lightning-kokkos-cuda
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+    libgomp1 \
+    python3 \
+    python3-pip \
+    python3-venv \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+ENV VIRTUAL_ENV=/opt/venv
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+COPY --from=build-kokkos-cuda /opt/kokkos /opt/kokkos
+COPY --from=build-wheel-lightning-kokkos-cuda /opt/pennylane-lightning/dist/ /
+RUN pip install --force-reinstall --no-cache-dir PennyLane_Lightning*.whl
 
-# Setup and build pennylane-lightning
+# Download and build Lightning-GPU release
+FROM base-build-cuda AS build-wheel-lightning-gpu
 WORKDIR /opt/pennylane-lightning
+ENV VIRTUAL_ENV=/opt/venv
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+RUN wget --progress=dot:giga https://github.com/PennyLaneAI/pennylane-lightning-gpu/archive/refs/tags/v${LIGHTNING_VERSION}.tar.gz
+RUN tar -zxf v${LIGHTNING_VERSION}.tar.gz && mv pennylane-lightning-gpu-${LIGHTNING_VERSION}/* /opt/pennylane-lightning
+RUN pip uninstall -y pennylane-lightning-gpu
+RUN pip install --no-cache-dir wheel custatevec-cu12
+RUN --mount=type=bind,source=requirements.txt,target=requirements.txt \
+    pip install --no-cache-dir -r requirements.txt
+RUN python setup.py build_ext --cuquantum=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum/lib')")
+RUN python setup.py bdist_wheel
+
+# Install python3 and setup runtime virtual env in CUDA-12-runtime image (includes CUDA runtime and math libraries)
+# Install lightning-kokkos CUDA backend
+FROM nvidia/cuda:12.2.0-runtime-ubuntu22.04 AS wheel-lightning-gpu
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+    libgomp1 \
+    python3 \
+    python3-pip \
+    python3-venv \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+ENV VIRTUAL_ENV=/opt/venv
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+RUN pip install --no-cache-dir custatevec-cu12
+COPY --from=build-wheel-lightning-gpu /opt/pennylane-lightning/dist/ /
+RUN pip install --no-cache-dir --force-reinstall PennyLane_Lightning*.whl
+
+# Install ROCm in build venv image
+FROM base-build-python as base-build-rocm
+RUN wget --progress=dot:giga https://repo.radeon.com/amdgpu-install/5.3/ubuntu/jammy/amdgpu-install_5.3.50300-1_all.deb
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+    "linux-headers-$(uname -r)" "linux-modules-extra-$(uname -r)" \
+    ./amdgpu-install_5.3.50300-1_all.deb \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+RUN amdgpu-install -y --accept-eula --usecase=rocm,hiplibsdk --no-dkms
```

Comment:
I don't think we want to accept EULAs in builds like this --- is it required? Is there another way?

Reply:
It is required for a non-interactive install. There is a [note](https://amdgpu-install.readthedocs.io/en/latest/install-overview.html) about it in the doc, saying "this option will prevent the proprietary packages or updates from being accessible directly with the system package manager". I could start from other ROCm containers to go around it, but I wanted to build all packages on the same footing as much as possible.

Reply:
Can ship from `rocm/dev-ubuntu-22.04`.